openfe/setup/atom_mapping/perses_mapper.py

```diff
@@ -0,0 +1,82 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+"""
+The MCS class from Perses shamelessly wrapped and used here to match our API.
+
+"""
+
+from openmm import unit
+
+#Test if Openeye license present! on class construction.
+from openff.toolkit.utils.exceptions import LicenseError
+from openff.toolkit.utils import toolkits
+
+from perses.rjmc.atom_mapping import AtomMapper, InvalidMappingException
+
+
+from .ligandatommapper import LigandAtomMapper
+
+
+class PersesAtomMapper(LigandAtomMapper):
+    unmap_partially_mapped_cycles: bool
+    preserve_chirality: bool
+
+    def __init__(self, full_cycles_only: bool = False,
+                 preserve_chirality: bool = True,
+                 use_positions: bool = True,
+                 coordinate_tolerance: float = 0.25 * unit.angstrom):
+        """
+        This class uses the perses code to facilitate the mapping of the
+        atoms of two molecules to each other.
+
+        Parameters
+        ----------
+        full_cycles_only: bool, optional
+            this option checks if on only full cycles of the molecules shall
+            be mapped, default: False
+        preserve_chirality: bool, optional
+             , default: True
+        use_positions: bool, optional
+            this option defines, if the
+        mapping_type: PersesMappingType, optional
+            how to calculate the mapping and amount of mappings,
+            default: PersesMappingType.best
+        coordinate_tolerance: float, optional
+            tolerance on how close coordinates need to be, such they
+            can be mapped, default: 0.25*unit.angstrom
+
+        """
+
+        self.unmap_partially_mapped_cycles = full_cycles_only
+        self.preserve_chirality = preserve_chirality
+        self.use_positions = use_positions
+        self.coordinate_tolerance = coordinate_tolerance
+
+        if (not toolkits.OPENEYE_AVAILABLE):
```

Comment:
Can you even import perses without openeye?

Reply:
yes you can :)

Reply:
so the if/raise is basically testing on the spot, if there is a license (similar how its not done in perses itself) 
@richardjgowers do you really want to fix this somewhere else? right now it looks to me to be very local in the perses functionality only.

Reply:
@richardjgowers Do you already have an idea, where I should move the license testing to?

Reply:
The license testing should be happening inside of perses.  We shouldn't be policing how dependencies are handling their dependencies.